src/io/config.cpp

```diff
@@ -45,7 +45,27 @@ void OverallConfig::Set(const std::unordered_map<std::string, std::string>& para
   if (boosting_type == BoostingType::kGBDT) {
     boosting_config = new GBDTConfig();
   }
-
+  
+  bool objective_type_multiclass = (objective_type == "multiclass");
+  int num_class_test = 1;
```

Comment:
@wxchan you can direct use "GetInt(params, "num_class", &num_class);" then test the conflict in the function "CheckParamConflict"